src/frontend/index.tsx

```diff
@@ -5,36 +5,42 @@ import jwtDecode from 'jwt-decode';
 import React from 'react';
 import ReactDOM from 'react-dom';
 import { addLocaleData, IntlProvider } from 'react-intl';
-import en from 'react-intl/locale-data/en';
-import es from 'react-intl/locale-data/es';
-import fr from 'react-intl/locale-data/fr';
 import { Provider } from 'react-redux';
 
 import { AppRoutesConnected } from './components/AppRoutesConnected/AppRoutesConnected';
 import { appData } from './data/appData';
 import { bootstrapStore } from './data/bootstrapStore';
-import translatedMessages from './translations/translation.json';
 import { DecodedJwt } from './types/jwt';
-import { localesMapping, TranslatedMessages } from './types/translation';
 // Load our style reboot into the DOM
 import { GlobalStyles } from './utils/theme/baseStyles';
 import { theme } from './utils/theme/theme';
 
 const store = bootstrapStore(appData);
-addLocaleData([...en, ...es, ...fr]);
 
 const decodedToken: DecodedJwt = jwtDecode(appData.jwt);
 
-const messages =
-  (translatedMessages as TranslatedMessages)[
-    localesMapping[decodedToken.locale]
-  ] || null;
+let localeCode = decodedToken.locale;
+if (localeCode.match(/^.*_.*$/)) {
+  localeCode = localeCode.split('_')[0];
+}
 
 // Wait for the DOM to load before we scour it for an element that requires React to render
-document.addEventListener('DOMContentLoaded', event => {
+document.addEventListener('DOMContentLoaded', async event => {
+  try {
```

Comment:
This logic belongs in a component so locale data can be fetched again if the language changes

Reply:
If language changes, all the application is reloaded, there is nothing allowing to change the language "dinamycally". It is the LTI request that is responsible to change the language.

Reply:
I understand, this ability is not necessary in Marsha as an LTI service.